commodore/catalog.py

```diff
@@ -175,6 +175,24 @@ def _kapitan_029_030_difffunc(
     before_text: str, after_text: str, fromfile: str = "", tofile: str = ""
 ) -> tuple[Iterable[str], bool]:
 
+    [diff, _] = _ignore_non_semantic_whitespace_difffunc(
+        before_text, after_text, fromfile, tofile
+    )
+
+    diff_lines = list(diff)
+
+    suppress_diff = not any(
+        _is_semantic_diff_kapitan_029_030(win)
+        for win in sliding_window(diff_lines[2:], 2)
+    )
+
+    return diff_lines, suppress_diff
+
+
+def _ignore_non_semantic_whitespace_difffunc(
+    before_text: str, after_text: str, fromfile: str = "", tofile: str = ""
+) -> tuple[Iterable[str], bool]:
+
     before_objs = sorted(yaml.safe_load_all(before_text), key=K8sObject)
     before_sorted_lines = yaml.dump_all(before_objs).split("\n")
```

Comment:
While this approach works to suppress non-semantic whitespace differences, it actually does a bunch more and will suppress reordering of objects in the output too. However, it doesn't look like Python's difflib has an option similar to `diff -b`, so this is probably the best we get.

Also, by using `yaml.dump_all()` without the custom dumper we use the wrong indentation for lists, which would generate inaccurate diffs if there's an actual change in a list combined with whitespace change.

Reply:
Maybe we should rename to `ignore-yaml-formatting-changes` or smth?

I'll use the same dumper

Reply:
Renaming it to `ignore-yaml-formatting` is probably a good idea to not mislead users.